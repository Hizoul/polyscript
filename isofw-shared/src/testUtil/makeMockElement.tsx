import * as React from "react"

const makeMockElement = (name: string) => {
  return (props: any) => (
    <div className={name}>
      {JSON.stringify(props, undefined, 2)}
    </div>
  )
  }

export default makeMockElement
