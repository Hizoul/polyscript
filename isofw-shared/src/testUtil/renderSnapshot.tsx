import * as React from "react"
const render = require("preact-render-to-json")

const renderSnapshot = (component: any, msg: string) => {
  expect(render(component)).toMatchSnapshot(msg)
}

export default renderSnapshot
