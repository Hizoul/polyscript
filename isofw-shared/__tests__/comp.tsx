import Comp from "isofw-shared/src/comp"
import makeMockElement from "isofw-shared/src/testUtil/makeMockElement"
import renderSnapshot from "isofw-shared/src/testUtil/renderSnapshot"
import * as React from "react"

test("Verify Component", () => {
  const mockEle = makeMockElement("mzmock")
  renderSnapshot(<Comp view={mockEle} />, "rendered")
})
