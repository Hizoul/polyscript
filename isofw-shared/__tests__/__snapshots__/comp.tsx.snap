// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify Component: rendered 1`] = `
<div
  class="mzmock"
>
  {
  "text": "shared",
  "children": []
}
</div>
`;
